UIforETW/UIforETWDlg.cpp

```diff
@@ -889,6 +898,54 @@ void CUIforETWDlg::StopTracingAndMaybeRecord(bool bSaveTrace)
 
 	ElapsedTimer saveTimer;
 	{
+		// If the user wants .NET stacks, and if they started the tracing after the process
+		// they are interested in, we need to capture rundown events so we can get names of
+		// methods that were jitted before we started tracing.
+		// https://msdn.microsoft.com/en-us/library/ff357718(v=vs.110).aspx
+		// This may take a moment, so if they want, they can uncheck the box before
+		// they stop tracing.
+		if (bSaveTrace && bDotNETStacks_)
+		{
+			std::wstring const clrRundownpath = GetCLRRundownFile();
+
+			{
+				ChildProcess rundownXperf(GetXperfPath());
+				std::wstring args = L" -start UIforETWSessionCLRRundown -on A669021C-C450-4609-A035-5AF59AF4DF18 -f \"" + clrRundownpath + L"\"";
+				rundownXperf.Run(bShowCommands_, L"xperf.exe" + args);
+			}
+			
+			// We've started the rundown process. We'll now wait for the size of the rundown etl to stop growing
+			// or for 120 seconds, whichever happens first.
+
+			{
+				auto const deadline = GetTickCount64() + 120000;
+				__int64 oldSize = 0;
+				while (GetTickCount64() < deadline)
+				{
+					Sleep(1000);
+					__int64 fileSize = GetFileSize(clrRundownpath);

```

Comment:
Sidenote: We should change this function so that it's obvious that we're not calling the [Windows API function](https://msdn.microsoft.com/en-us/library/windows/desktop/aa364955.aspx).